core/asset/discovery.go

```diff
@@ -12,6 +12,10 @@ type DiscoveryRepository interface {
 	Search(ctx context.Context, cfg SearchConfig) (results []SearchResult, err error)
 	Suggest(ctx context.Context, cfg SearchConfig) (suggestions []string, err error)
 	GroupAssets(ctx context.Context, cfg GroupConfig) (results []GroupResult, err error)
+	Clone(ctx context.Context, indexName, clonedIndexName string) error
+	UpdateAlias(ctx context.Context, indexName, alias string) error
+	DeleteByIndexName(ctx context.Context, indexName string) error
+	UpdateIndexSettings(ctx context.Context, indexName string, body string) error
```

Comment:
Let's avoid exposing this and keep as internal implementation, instead, expose the SyncAssets operation

Reply:
This might be useful for folks who might want to update index settings, through a script which has compass code as dependency. WDYT ?

Reply:
Currently there's no use case yet to call this apart from sync logic, we can consider exposing this later if there's more use case. For now let me make it private